bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/SitemapResource.java

```diff
@@ -210,13 +210,17 @@ public Response getPageData(@Context HttpHeaders headers,
             }
         }
 
+        boolean timeout = false;
         if (headers.getRequestHeader("X-Atmosphere-Transport") != null) {
             // Make the REST-API pseudo-compatible with openHAB 1.x
             // The client asks Atmosphere for server push functionality,
             // so we do a simply listening for changes on the appropriate items
-            blockUnlessChangeOccurs(sitemapname, pageId);
+            // The blocking has a timeout of 30 seconds. If this timeout is reached,
+            // we notice this information in the response object.
+            timeout = blockUnlessChangeOccurs(sitemapname, pageId);
         }
-        Object responseObject = getPageBean(sitemapname, pageId, uriInfo.getBaseUriBuilder().build(), locale);
+        PageDTO responseObject = getPageBean(sitemapname, pageId, uriInfo.getBaseUriBuilder().build(), locale);
+        responseObject.timeout = timeout;
```

Comment:
Imho it would be cleaner to add `timeout` as a new parameter to `getPageBean` - this should be safe as it is a only a private method.

Reply:
Ok I will change that.